### ISSUE TITLE
support `Generic` bound evaluation on `typing.Self`

### ISSUE THREAD

**Issue by CoderJoshDK** [astral-sh/ty#1335]
Given the following python code

```py
from typing import Generic, Self, TypeVar

class Base: ...

V = TypeVar("V", bound="Base")
# V = TypeVar("V", bound="Base", covariant=True) # also fails


class Item(Generic[V]): ...

class Fine(Base): ...

class Problem(Base):
    item: Item[Self]
    good: Item[Fine]

    def __init__(self) -> None:
        super().__init__()

        self.item = Item()
        self.good = Item()
```

`ty` reports:

```sh
> uvx ty check
WARN ty is pre-release software and not ready for production use. Expect to encounter bugs, missing features, and fatal errors.
Checking ------------------------------------------------------------ 1/1 files                                                                                                                error[invalid-argument-type]: Argument to class `Item` is incorrect
  --> main.py:18:11
   |
17 | class Problem(Base):
18 |     item: Item[Self]
   |           ^^^^^^^^^^ Expected `Base`, found `typing.Self`
19 |     good: Item[Fine]
   |
info: rule `invalid-argument-type` is enabled by default

Found 1 diagnostic
```

As you can see, with `good` // `Fine`, `ty` correctly can determine that `Fine` is bound to `Base` but it can not evaluate `Self` as being bound to `Base`

<details><summary>Version</summary>
<p>

```sh
> uvx ty --version
ty 0.0.1-alpha.22 (2f3190d09 2025-10-10)
```

</p>
</details> 

<sub>Potentially related (but inverted) is #1124.</sub>

**Comment by carljm** [comment]
Thanks for the report! Although the presentation is a bit different, the root cause here is exactly the same as #1124; we don't have any support yet for `Self` as an attribute annotation.
